app/controllers/sessions_controller.rb

```diff
@@ -25,9 +25,15 @@ def create
   end
 
   def twitter
-    # raise request.env["omniauth.auth"].to_yaml
+    #raise request.env["omniauth.auth"].to_yaml
     auth = request.env["omniauth.auth"]
-    user = User.where(uid: auth["uid"]).first || User.from_twitter(auth)
+    if session[:user_id] #MOREMORE how to test if a user is signed in here?
+      user = current_user
+      user.uid = auth["uid"].first
+      user.name = auth["screen_name"].first
```

Comment:
that's not correct.  You can use `auth.info.nickname` for the nickname. Also, make sure you call `#save` so that the data is actually saved.

Also, just `auth.uid` is fine, don't call `#first` on it.